### PR TITLE
Module Component

### LINKED ISSUE #101: Modules Frontend Component
image
^ here is the image of the view to create 

In order to get a sense of the API routes, take a look at #30 . It has a good overview of how to post and add modules and what they contain (items) into the database. Adding a module and item shouldnt be too bad - both just require interacting with some POST routes. 

However, the best way to go about this is probably creating a component called `Module` that renders the dropdown and all the items within it. Within the modules component, you can make a call to the backend to get all the links that belong within that module. This component should also handle the adding, deleting, and editing of the individual items. Creating an item should just have a pop up that looks like the create module pop up, but with name and link instead. The API does not have support to editing modules, so if you could add that in, that would be great. Should just be a PATCH to `/api/module/[module_id]/item/[item_id]` with the following at the body:
```
type EditItem = {
title?: string, 
link?: string 
}
```

Finally, make sure this modules parameter takes in some prop called `enableEditing`. Only allow editing functions to appear when this is set to `true`

In the overall page, we can make a call to `GET /api/class/[id]/modules
` to get all the modules that belong to that class. Then, simply render the `Modules` components in the order specified. 

We can add the ability to move a module up and down. As soon as someone asks a module to move up, we can fire off a call to the database with the new order. We can do this by having an array of moudle id’s in the order they are currently being displayed, and having a function that edits the order when one of the move up/down buttons are called. Make sure to keep track of the fact that the topmost module can’t move up and the bottom most module cant move down (maybe have a min/max that you check).

### PR DESCRIPTION
**Pull request by nishantbalaji**
### Administrative Info

Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Github Issue]-[3-4 word description separated by dashes]`.

Remade from #109 

### What issue(s) does this branch close?
Closes #101 

### Changes

What changes did you make?

- Added a Module view with dropdowns showing the modules

### Testing

How did you confirm your changes worked?

- Test with dummy data and make sure they dsplay

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Module View |
| ----------------------- |
| image  |

### CODE REVIEW

File: pages/api/module/[id]/item/[item_id].ts [hunk 1]
```diff
@@ -72,6 +74,24 @@ export const deleteItemHandler: NextApiHandler = async (
         return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json("Internal Server CustomError");
```
Comment (Anshul-Birla) [inline pages/api/module/[id]/item/[item_id].ts:74]: lets delete the custom error?

File: package.json [hunk 2]
```diff
@@ -58,6 +58,7 @@
     "pino-pretty": "^9.1.1",
     "postgres-migrations": "^5.3.0",
     "react": "17.0.2",
+    "react-bootstrap": "^2.6.0",
```
Comment (Anshul-Birla) [inline package.json:61]: not sure how i feel about using react-bootstrap. Its such a heavy library, and this app is kinda heavy to begin with

File: components/SingleClassView/Module/Module.tsx [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import Fade from "@mui/material/Fade";
+
+type ModuleProps = {
+  id: string;
+  enableEditing: boolean;
+};
+
+// eslint-disable-next-line
```
Comment (Anshul-Birla) [inline components/SingleClassView/Module/Module.tsx:16]: why are these disabled for the linter?

File: components/SingleClassView/Module/Module.tsx [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+  const handleMoveUp: VoidFunction = () => {
+    if (position > 0) {
+      setPosition(position - 1);
+    }
+    handleClose();
+  };
+
+  const handleMoveDown: VoidFunction = () => {
+    if (position < numModules) {
+      setPosition(position + 1);
```
Comment (Anshul-Birla) [inline components/SingleClassView/Module/Module.tsx:201]: you have to also reconfigure everything else above and below for this to work correctly

Comment (Anshul-Birla) [reply components/SingleClassView/Module/Module.tsx:201]: Same for move above

Comment (Anshul-Birla) [reply components/SingleClassView/Module/Module.tsx:201]: You will need to store this in an array so you can handle the move up and down in an ordered fashion

Comment (Anshul-Birla) [reply components/SingleClassView/Module/Module.tsx:201]: Also, you need to make a PATCH call with the modules to save the updated data in the backend

File: components/SingleClassView/Module/Module.tsx [hunk 5]
```diff
@@ -0,0 +1,400 @@
+import React, { useContext, useEffect, useState } from "react";
```
Comment (Anshul-Birla) [inline components/SingleClassView/Module/Module.tsx:1]: github stopped rendering this dif, so i think this file is too big. Lets split these into different components, I can do it if u would like (jetbrains makes it easy af)

File: components/League/ModuleCard.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const ModuleCard: React.FC = () => {
+  return <div>Hi</div>;
+};
```
Comment (Anshul-Birla) [inline components/League/ModuleCard.tsx:5]: ?

**Review by Anshul-Birla — CHANGES_REQUESTED** [review]
Some pointers: 
- the save button does not work when editing a item 
- the module items should be links. Rn, we cant access the links lol
- Move up/down does not work 
- I get a 404 when trying to create a module

File: package.json (hunk 2 above)
Comment (Anshul-Birla) [reply package.json:61]: why do you need it?

File: components/League/ModuleCard.tsx (hunk 6 above)
Comment (nishantbalaji) [reply components/League/ModuleCard.tsx:5]: lmao i totally forgot about this

File: components/SingleClassView/Module/Module.tsx (hunk 4 above)
Comment (nishantbalaji) [reply components/SingleClassView/Module/Module.tsx:211]: addressed in slack

File: components/SingleClassView/Module/Module.tsx (hunk 5 above)
Comment (nishantbalaji) [reply components/SingleClassView/Module/Module.tsx:1]: please do if you can <3

**Review by Anshul-Birla — APPROVED** [review]

**Pull request merged**